sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -614,7 +614,7 @@ def download_blob(
         self, offset: Optional[int] = None,
         length: Optional[int] = None,
         *,
-        encoding: Optional[str] = None,
+        encoding: None = None,
```

Comment:
I think this was the issue with our overload. If my understanding is correct, if encoding is str --> will return a `StorageStreamDownloader[str]`. If encoding is unset, will be `StorageStreamDownloader[bytes]`.

So our issue was that previously we had the bytes case have an `encoding: Optional[str]` type-hint, which covers the domain of regular `str`. Thus the overload wasn't able to "gate" properly and know which one should result to which return type. 

BUT I saw we didn't have this issue in one of the container client's, and that cleared everything up 😄